### PR TITLE
dev: Add builder pattern for the machine object in tests

### PR DESCRIPTION
**Pull request by edisontim**
## Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [X] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?
Resolves: #546

## What is the new behavior?

Add builder pattern for Machine object in tests

## Does this introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

**Comment by enitrat** [comment]
agreed - the way it has been implemented just abstracts it but doesn't make it simpler!

**Comment by edisontim** [comment]
Ok I'll remove the Director object from there then :)

**Comment by edisontim** [comment]
Makes sense to me to also get rid of ```setup_execution_context_with_bytecode``` and all the others, and for ```test_execution_context_read_code``` and other tests like that to create a machine to run the tests. What do you think?

**Comment by Eikix** [comment]
> Makes sense to me to also get rid of `setup_execution_context_with_bytecode` and all the others, and for `test_execution_context_read_code` and other tests like that to create a machine to run the tests. What do you think?

I think you're right

**Comment by edisontim** [comment]
@enitrat @enitrat Is this what you had in mind?

File: crates/evm/src/tests/test_execution_context.cairo [hunk 1]
```diff
@@ -56,8 +55,9 @@ fn test_call_context_new() {
 #[test]
 #[available_gas(500000)]
 fn test_execution_context_new() {
+    let mut machine = MachineBuilderImpl::new_with_presets().build();
     // Given
-    let call_ctx = setup_call_context();
+    let call_ctx = machine.current_ctx.unbox().call_ctx();
```
Comment (enitrat) [inline crates/evm/src/tests/test_execution_context.cairo:60]:
```suggestion
    let call_ctx = machine.call_ctx();
```

File: crates/evm/src/tests/test_execution_context.cairo [hunk 2]
```diff
@@ -6,10 +7,8 @@ use evm::context::{
 use evm::memory::{Memory, MemoryTrait};
 use evm::model::{Event, Address};
```
Comment (enitrat) [inline crates/evm/src/tests/test_execution_context.cairo:8]:
```suggestion
use evm::machine::{MachineCurrentContextTrait};
```

**Review by enitrat — COMMENTED** [review]
yep looks good now!

File: crates/evm/src/tests/test_execution_context.cairo [hunk 3]
```diff
@@ -97,10 +97,11 @@ fn test_execution_context_new() {
 }
 
 #[test]
-#[available_gas(100000)]
+#[available_gas(1000000)]
 fn test_execution_context_stop_and_revert() {
     // Given
-    let mut execution_context = setup_execution_context();
+    let mut machine = MachineBuilderImpl::new_with_presets().build();
+    let mut execution_context = machine.current_ctx.unbox();
```
[the remark this answers is not in the document]
Comment (edisontim) [reply crates/evm/src/tests/test_execution_context.cairo:104]: You mean remove ```setup_execution_context``` everywhere or getting the ```current_ctx``` with a function? Can't find the function anyways even with the trait

[the remark this answers is not in the document]
Comment (enitrat) [reply crates/evm/src/tests/test_execution_context.cairo:104]: ah sorry my bad - it looks like we don't have a `current_ctx` getter in `machine`. could you add one ?

**Comment by enitrat** [comment]
all good, please rebase!

**Comment by edisontim** [comment]
Done :) !

**Review by enitrat — APPROVED** [review]
@Eikix LGTM

**Pull request merged**